torch/csrc/autograd/engine.cpp

```diff
@@ -386,7 +386,11 @@ void GraphTask::set_exception(
       fn->metadata()->print_stack();
     }
     has_error_ = true;
-    future_result_->setError(e.what());
+    if (!future_result_->completed()) {
+      future_result_->setError(e.what());
+    } else {
+      TORCH_INTERNAL_ASSERT(future_result_->hasError());
```

Comment:
Can that happen due to a race condition if two thread set the error at the same time?

Reply:
Ho the lock was outside the online view.. Sounds good !